pyspark_huggingface/huggingface_sink.py

```diff
@@ -265,25 +291,43 @@ def flush(writer: pq.ParquetWriter):
         return HuggingFaceCommitMessage(additions=additions)
 
     def commit(self, messages: List[HuggingFaceCommitMessage]) -> None:  # type: ignore[override]
-        import math
+        api = self._get_api()
 
-        api = self.get_api()
         operations = [
             addition for message in messages for addition in message.additions
         ]
```

Comment:
I checked the code at https://huggingface.co/docs/hub/en/datasets-spark#write and found we actually don't need 2 steps !

This should set the correct name in a single step by changing the `addition.path_in_repo` in `commit()`:

```suggestion
        operations = [
            addition for message in messages for addition in message.additions
        ]
        num_shards = len(operations)
        for shard_idx, addition in enumerate(operations):
            addition.path_in_repo = f"data/{self.split}-{shard_idx:05d}-of-{num_shards:05d}.parquet"
```

Indeed the `addition.path_in_repo` is only used when committing, and is not used at all when doing `preupload_lfs_files()`

Reply:
> we actually don't need 2 steps

This is the case when there's no files to delete or rename, e.g. when writing to an empty dataset repo. Otherwise, in append mode, we first need to rename existing files to avoid overwriting the files which are supposed to be kept.

In overwrite mode I think only 1 commit is needed if CommitOperationAdd lets us overwrite files. I'll check whether that works.

Reply:
Updated the overwrite logic to have 1 commit not 2